Thanks for the pagination PR @vedikathalkar! Useful scalability fix 

Two things to fix:

1. Broken test — update assertions in backend/tests/test_api.py to expect the new paginated dictionary structure instead of a plain list:
   AssertionError: {'items': [], 'total': 0, ...} != []

2. Add trailing newline at end of backend/app/main.py

Also please verify frontend components handle the new paginated response shape. Let me know once fixed!

--- a/backend/tests/test_api.py
+++ b/backend/tests/test_api.py
@@ -379,7 +379,12 @@ class PrepIQApiTestCase(unittest.TestCase):
 
         jobs = self.client.get(f"/api/users/{user_id}/jobs", headers=headers)
         self.assertEqual(jobs.status_code, 200, jobs.text)
-        self.assertEqual(jobs.json(), [])
+        jobs_payload = jobs.json()
+        self.assertEqual(jobs_payload["items"], [])
+        self.assertEqual(jobs_payload["total"], 0)
+        self.assertEqual(jobs_payload["page"], 1)
+        self.assertEqual(jobs_payload["limit"], 20)
+        self.assertEqual(jobs_payload["total_pages"], 0)
 
         delete_session = self.client.delete(
             f"/api/users/{user_id}/sessions/{session_payload['id']}",
@@ -389,7 +394,12 @@ class PrepIQApiTestCase(unittest.TestCase):
 
         sessions = self.client.get(f"/api/users/{user_id}/sessions", headers=headers)
         self.assertEqual(sessions.status_code, 200, sessions.text)
-        self.assertEqual(sessions.json(), [])
+        sessions_payload = sessions.json()
+        self.assertEqual(sessions_payload["items"], [])
+        self.assertEqual(sessions_payload["total"], 0)
+        self.assertEqual(sessions_payload["page"], 1)
+        self.assertEqual(sessions_payload["limit"], 20)
+        self.assertEqual(sessions_payload["total_pages"], 0)
 
         mocks_after_delete = self.client.get(
             f"/api/users/{user_id}/mocks", headers=headers

--- a/src/lib/store.ts
+++ b/src/lib/store.ts
@@ -138,6 +138,22 @@ export interface JobApplication {
   updatedAt: string;
 }
 
+export interface PaginatedJobApplications {
+  items: JobApplication[];
+  total: number;
+  page: number;
+  limit: number;
+  total_pages: number;
+}
+
+export interface PaginatedInterviewSessions {
+  items: InterviewSession[];
+  total: number;
+  page: number;
+  limit: number;
+  total_pages: number;
+}
+
 export interface CreateInterviewSessionInput {
   jobTitle: string;
   company: string;
@@ -179,6 +195,14 @@ function setSession(session: AuthSession | null) {
 // Legacy useProtectedResource hook removed in favor of TanStack React Query.
 
 export function getMockAttemptItems(payload: MockAttempt[] | PaginatedMockAttempts): MockAttempt[] {
+  return Array.isArray(payload) ? payload : payload.items;
+}
+
+export function getJobApplicationItems(payload: JobApplication[] | PaginatedJobApplications): JobApplication[] {
+  return Array.isArray(payload) ? payload : payload.items;
+}
+
+export function getInterviewSessionItems(payload: InterviewSession[] | PaginatedInterviewSessions): InterviewSession[] {
   return Array.isArray(payload) ? payload : payload.items;
 }
 
@@ -303,7 +327,10 @@ export function useInterviewSessions(userId: string | undefined) {
   const queryClient = useQueryClient();
   const sessionsQuery = useQuery<InterviewSession[]>({
     queryKey: ["interviewSessions", userId],
-    queryFn: () => apiRequest<InterviewSession[]>(`/api/users/${userId}/sessions`),
+    queryFn: async () => {
+      const payload = await apiRequest<InterviewSession[] | PaginatedInterviewSessions>(`/api/users/${userId}/sessions`);
+      return getInterviewSessionItems(payload);
+    },
     enabled: !!userId,
   });
 
@@ -389,7 +416,10 @@ export function useJobApplications(userId: string | undefined) {
   const queryClient = useQueryClient();
   const jobsQuery = useQuery<JobApplication[]>({
     queryKey: ["jobApplications", userId],
-    queryFn: () => apiRequest<JobApplication[]>(`/api/users/${userId}/jobs`),
+    queryFn: async () => {
+      const payload = await apiRequest<JobApplication[] | PaginatedJobApplications>(`/api/users/${userId}/jobs`);
+      return getJobApplicationItems(payload);
+    },
     enabled: !!userId,
   });
 
@@ -446,4 +476,3 @@ export function useJobApplications(userId: string | undefined) {
   const jobs = jobsQuery.data ?? [];
   return { jobs, addJob, updateJob, deleteJob, jobsError: jobsQuery.error instanceof Error ? jobsQuery.error.message : null, jobsLoading: jobsQuery.isLoading };
 }
-

